File: api/src/main/java/com/api/app/entity/PayInterfaceLog.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.io.Serializable;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("PayInterfaceLog")
@Getter
@Setter
public class PayInterfaceLog implements Serializable {
    private static final long serialVersionUID = 6789012345678901234L;

    @Schema(description = "결제인터페이스번호")
    private String payInterfaceNo;

    @Schema(description = "회원번호")
    private String memberNo;

    @Schema(description = "결제번호")
    private String payNo;

    @Schema(description = "로그유형코드")
    private String payLogCode;

    @Schema(description = "요청JSON")
    private String requestJson;

    @Schema(description = "응답JSON")
    private String responseJson;
}

File: api/src/main/kotlin/com/api/app/entity/PayInterfaceLog.kt
package com.api.app.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "PAY_INTERFACE_LOG")
class PayInterfaceLog : SystemEntity() {

    @Id
    @Column(name = "PAY_INTERFACE_NO", length = 15, nullable = false)
    var payInterfaceNo: String = ""

    @Column(name = "MEMBER_NO", nullable = false, length = 15)
    var memberNo: String = ""

    @Column(name = "PAY_NO", nullable = false, length = 15)
    var payNo: String = ""

    @Column(name = "PAY_LOG_CODE", nullable = false, length = 3)
    var payLogCode: String = ""

    @Column(name = "REQUEST_JSON", columnDefinition = "TEXT")
    var requestJson: String? = null

    @Column(name = "RESPONSE_JSON", columnDefinition = "TEXT")
    var responseJson: String? = null
}
